### PR TITLE
refactor: fix nil pointer when glog is removed & update old `-v=` to `-log-level=`

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

- fix nil pointer when glog is removed in flags.go to prepare for full glog removal in the `healthcheck` package soon by removing a check for`logtostderr` and `include_year` flags as `logtostderr` was included with glog and `include_year` has been removed. 
- update old `-v=` to `-log-level=` as it was only used for glog which has now been switched out and `-v=` has no longer been used since #6520  

What logtostderr flag was doing in glog 
```
logtostderr (bool, default=false)
Log messages to stderr instead of logfiles.
```

Helm-chart will need to be updated to v2.0.0 as we have a breaking change changing from `-v` to `-log-level` so this needs to be done

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/daemon-set/nginx-ingress.yaml [hunk 1]
```diff
@@ -91,7 +91,8 @@ spec:
           - -report-ingress-status
           - -external-service=nginx-ingress
          #- -default-server-tls-secret=$(POD_NAMESPACE)/default-server-secret
-         #- -v=3 # Enables extensive logging. Useful for troubleshooting.
+         #- -log-level=debug # Enables extensive logging. Useful for troubleshooting.
```
Comment (jjngx) [inline deployments/daemon-set/nginx-ingress.yaml:94]: How about to add available level options (same as L95 for `-log-format`)?

**Review by jjngx — APPROVED** [review]
🚀

**Review by ADubhlaoich — APPROVED** [review]
LGTM, though the PR is largely not docs-based.

**Comment by j1m-ryan** [comment]
can you update the docs here https://github.com/nginxinc/kubernetes-ingress/blob/main/site/content/logging-and-monitoring/logging.md?

**Review by j1m-ryan — APPROVED** [review]

**Pull request merged**